### ISSUE TITLE
Using in a multi-tenant environment

### ISSUE THREAD

**Issue by arashster** [RIPAGlobal/scimitar#57]
This is not really a bug but a question/request for clarification.

Is there a sample/guideline to show how to use this gem in a multitenant environment?

How can you set up the  '`token_authenticator`' proc in `EngineConfiguration.new` to match, for example, the request hostname or a route parameter? Is this even the right place to do it, or would you do that in the controller?

Thanks for the clarification.

**Comment by pond** [comment]
There's no clean answer. Microsoft (Azure AD) has its own ideas about tenancy, and I'm not 100% clear on what they want o do with SCIM there. In our own application's case, tenancy is asserted by subdomain - `foo.ripaglobal.com` means the tenancy name is `foo`. Tenant `foo`'s corporate SCIM setup is configured to use `foo.ripaglobal.com` as the API endpoint, so whenever we are called, we can find the tenant and assert validity at that point. We do this via engine configuration in `config/initializers/scimitar.rb`:

```ruby
Rails.application.config.to_prepare do
  Scimitar.engine_configuration = Scimitar::EngineConfiguration.new({

    application_controller_mixin: Module.new do
      def self.included(base)
        base.class_eval do
          include CommonSetup # This is a controller concern we wrote and reuse in a few places

          skip_before_action    :verify_authenticity_token
          prepend_before_action :common_assert_tenancy! # This is in CommonSetup & examines request subdomain 
        end
      end
    end,
end
```

**Comment by pond** [comment]
For the sake of archives and web searches, mostly:

There's a followup here that provides an alternative. A PR submission in https://github.com/RIPAGlobal/scimitar/pull/71 addresses tenancy another way, via URL route parameters rather than subdomains. This might be of interest. It's almost over the line so should be merged pretty quickly.

**Comment by pond** [comment]
Final follow-up - the PR mentioned above has been merged; Scimitar v2.6.0 or later (Rails 7) / v1.7.0 or later (Rails 6) include this change.
